server/src/main/java/keywhiz/service/daos/SecretSeriesDAO.java

```diff
@@ -119,21 +125,78 @@ long createSecretSeries(
     r.setUpdatedat(now);
     r.setType(type);
     r.setRowHmac(rowHmac);
-    if (generationOptions != null) {
-      try {
-        r.setOptions(mapper.writeValueAsString(generationOptions));
-      } catch (JsonProcessingException e) {
-        // Serialization of a Map<String, String> can never fail.
-        throw Throwables.propagate(e);
-      }
-    } else {
-      r.setOptions("{}");
-    }
+    r.setOptions(getOptionsField(generationOptions));
+
     r.store();
 
     return r.getId();
   }
 
+  public long createDeletedSecretSeries(
```

Comment:
This only exists for testing, but I can remove it in my next PR. It's helpful right now because there's no other method create a row in the deleted_secrets table. After my next PR, I can put a row in the deleted_secrets table by just creating a secret and then deleting it, so I won't need this method.

I could refactor this so that this creation function is in test code, but that would be a lot more complicated because this function depends on member variables that are instantiated via dependency injection, and I would have to set all that up in my test class. Since I'm about to remove this function anyway, I don't think it's worth it.